### PR TITLE
ci(NODE-5653): remove vars from ci config

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- remove use of vars
- remove mongosh tests
- move vars to expansions

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

vars are being changed to not exist after functions this avoids any silent config issues.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/ci_matrix_constants.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const versions = [
   { codeName: 'hydrogen', versionNumber: 18 },
   { codeName: 'iron', versionNumber: 20 }
 ];
-const NODE_VERSIONS = versions.map(({ versionNumber }) => versionNumber).sort();
+const NODE_VERSIONS = versions.map(({ versionNumber }) => versionNumber).sort((a, b) => a - b);
```
Comment (baileympearson) [inline .evergreen/ci_matrix_constants.js:8]: why is this change necessary?

**Review by baileympearson — CHANGES_REQUESTED** [review]

File: .evergreen/ci_matrix_constants.js (hunk 1 above)
Comment (nbbeeken) [reply .evergreen/ci_matrix_constants.js:8]: It isn't the atlas search tests started failing b/c they were falling back to node 14 but they were set to use "latest lts" so I thought there was something wrong with the sorting. But they just used the wrong variable name NODE_LTS_NAME vs NODE_LTS_VERSION. 

However, without this comparator, sort doesn't perform what you would expect:
```
> [1,3,2,100].sort()
[ 1, 100, 2, 3 ]
```
I figure why not leave behind the correct thing so it never bites us

Comment (baileympearson) [reply .evergreen/ci_matrix_constants.js:8]:
```
> [1,3,2,100].sort()
[ 1, 100, 2, 3 ]
```
TIL! thanks.

**Review by baileympearson — APPROVED** [review]
